sim-lib/src/lib.rs

```diff
@@ -554,9 +570,25 @@ impl Simulation {
         );
 
         // Next, we'll spin up our actual producers that will be responsible for triggering the configured activity.
-        self.dispatch_producers(activities, consumer_channels, &mut tasks)
+        // The producers will use their own JoinSet so that the simulation can be shutdown if they all finish.
+        let mut producer_tasks = JoinSet::new();
```

Comment:
Nice simple approach ✨ 

Very nifty that we don't need to wait on `producer_tasks` at the end of the simulation because they're represented by our high level `tasks` set! 

@enigbe possibly something that we could look at for `track_payment`?